MinecraftClient/ChatBots/AutoAttack.cs

```diff
@@ -40,14 +55,53 @@ public override void Update()
                 attackCooldownCounter = attackCooldown;
                 if (entitiesToAttack.Count > 0)
                 {
-                    SendAnimation(Inventory.Hand.MainHand); // Arm animation
-                    foreach (KeyValuePair<int, Entity> entity in entitiesToAttack)
+                    if (!singleMode)
+                    {
+                        foreach (KeyValuePair<int, Entity> entity in entitiesToAttack)
+                        {
+                            // check that we are in range once again.
+                            bool shouldAttack = handleEntity(entity.Value);
```

Comment:
How about renaming `handleEntity` to `shouldAttackEntity`? This way no need to use an explicit variable:
```C#
foreach (KeyValuePair<int, Entity> entity in entitiesToAttack)
    if (shouldAttackEntity(entity.Value))
        InteractEntity(entity.Key, 1); // hit the entity!
```
(You can add braces `{` `}` if you prefer, it's up to you)

Reply:
Okay.
_Actually not me implemented this part_

Reply:
Ah, sorry. Did not check the git blame 😄